external/kafka-0-10/src/test/scala/org/apache/spark/streaming/kafka010/KafkaRDDSuite.scala

```diff
@@ -44,20 +44,25 @@ class KafkaRDDSuite extends SparkFunSuite with BeforeAndAfterAll {
   private var sc: SparkContext = _
 
   override def beforeAll {
+    super.beforeAll()
     sc = new SparkContext(sparkConf)
     kafkaTestUtils = new KafkaTestUtils
     kafkaTestUtils.setup()
   }
 
   override def afterAll {
-    if (sc != null) {
-      sc.stop
-      sc = null
-    }
-
-    if (kafkaTestUtils != null) {
-      kafkaTestUtils.teardown()
-      kafkaTestUtils = null
+    try {
+      if (sc != null) {
+        sc.stop
+        sc = null
+      }
+
+      if (kafkaTestUtils != null) {
+        kafkaTestUtils.teardown()
+        kafkaTestUtils = null
+      }
+    } finally {
+      super.afterAll()
```

Comment:
@kiszk Can sc.stop throw an exception ? If so, shouldn't we attempt to do a teardown first before calling super.afterAll() ?

Reply:
I'm confused by your comment, it looks like we try and stop the spark context first. Are you suggesting that we want to tear down kafka test utils even if the Spark `stop` context fails?

Reply:
@holdenk Please correct me on this one, in my understanding, we do 3 distinct things in the afterAll() method.
```
1) sc.stop 
2) kafkaTestUtils.teardown() 
3) super.afterAll()
```
Currently, even if we fail in (1) or (2) , we will always do (3) as its done in the finally block. My question was , if we fail at (1) , should we do (2) and (3) ? I don't know about the details of the test suite to know for sure if this matters or not. I was just curious.

Reply:
`sc.stop` can throw an exception. Thus, I updated this using nested `try-finally`.